src/gui/layertree/qgslayertreeview.cpp

```diff
@@ -406,7 +406,10 @@ QgsMapLayer *QgsLayerTreeView::layerForIndex( const QModelIndex &index ) const
 
 QgsLayerTreeNode *QgsLayerTreeView::currentNode() const
 {
-  return index2node( selectionModel()->currentIndex() );
+  if ( auto selectModel = selectionModel() )
```

Comment:
Qgis coding convention is to avoid auto for trivial types like this -- can you put the explicit type please?